aten/src/ATen/native/Memory.cpp

```diff
@@ -3,14 +3,24 @@
 #include <ATen/NativeFunctions.h>
 #include <ATen/TensorUtils.h>
 #include <ATen/detail/CUDAHooksInterface.h>
+#include <ATen/detail/HPUHooksInterface.h>
 #include <c10/util/Exception.h>
 #include <c10/core/Storage.h>
 
 namespace at {
 namespace native {
 
 bool is_pinned(const Tensor& self) {
-  return detail::getCUDAHooks().isPinnedPtr(self.storage().data());
+  if (detail::getCUDAHooks().hasCUDA()) {
+    return detail::getCUDAHooks().isPinnedPtr(self.storage().data());
+  }
+  else if (detail::getHPUHooks().hasHPU()) {
+    return detail::getHPUHooks().isPinnedPtr(self.storage().data());
+  }
```

Comment:
This logic looks wrong. In the hypothetical situation where you have both CUDA and HPU loaded, you will never go to getHPUHooks even if the tensor is pinned for HPU. I feel that @ngimel and I have discussed in the past about how `is_pinned` may need to accept a DeviceType, so that you can ask the question "pinned for CUDA/HPU"

Reply:
Yes, agreed. Is there a plan to add the modification DeviceType for in_pinned soon?